src/sentry/sentry_apps/token_exchange/grant_exchanger.py

```diff
@@ -65,10 +65,10 @@ def _validate(self) -> None:
         Validator(install=self.install, client_id=self.client_id, user=self.user).run()
 
         if not self._grant_belongs_to_install() or not self._sentry_app_user_owns_grant():
-            raise APIUnauthorized("Forbidden grant")
+            raise SentryAppIntegratorError(message="Forbidden grant")
 
         if not self._grant_is_active():
-            raise APIUnauthorized("Grant has already expired")
+            raise SentryAppIntegratorError("Grant has already expired", status_code=403)
```

Comment:
`APIUnauthorized` has status code 401

```suggestion
            raise SentryAppIntegratorError("Grant has already expired", status_code=401)
```